src/main/java/duke/commands/DeadlineCommand.java

```diff
@@ -0,0 +1,69 @@
+package duke.commands;
+
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.tasks.Deadline;
+
+/**
+ * Adds a deadline task to the task list.
+ */
+public class DeadlineCommand extends Command {
+    /**
+     * The command word to identity DeadlineCommand instance.
+     */
+    public static final String COMMAND_WORD = "deadline";
+
+    /**
+     * Guide on how to use this command word.
+     */
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + " <description> /by <date> - add a task to be completed by <date> in yyyy/MM/dd HHmm (24-hour format)\n"
+            + "   Example: " + COMMAND_WORD + " project submission /by 2021/08/30 2359";
+
+    private String userCommand;
+
+    /**
+     * Instantiates DeadlineCommand object.
+     *
+     * @param userCommand Full user input.
+     */
+    public DeadlineCommand(String userCommand) {
+        super();
+        this.userCommand = userCommand;
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        try {
+            int byIndex = userCommand.indexOf("/by");
+            if (userCommand.length() < 9 || byIndex <= 9) {
```

Comment:
Should this magic number be moved to top by using named constant? 🤔
And I noticed the same issue in several other places below.